pkgs/leak_tracker_flutter_test/test/end_to_end/leak_tracking_config_test/leak_tracking_config_test.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+
+import '../../test_infra/flutter_classes.dart';
+import '../../test_infra/leak_tracking_in_flutter.dart';
+
+const test1TrackingOn = 'test1, tracking-on';
+const test2TrackingOff = 'test2, tracking-off';
+const test3TrackingOnWithStackTrace = 'test3, tracking-on, with stack trace';
+
+/// For these tests `expect` for found leaks happens in flutter_test_config.dart.
+void main() {
+  testWidgetsWithLeakTracking(test1TrackingOn, (widgetTester) async {
```

Comment:
It might be better to have the test names here in strings instead of in variables. That way it will be easier to read what is being tested without needing to jump around.
Example: "starting phases with tracking-on"

Reply:
+1, if we're not generating the names programmatically, we should just use string literals in `testWidgets(...)`